RecoEgamma/EgammaTools/interface/Spot.h

```diff
@@ -0,0 +1,37 @@
+#ifndef SPOT_H
+#define SPOT_H
+#include "DataFormats/DetId/interface/DetId.h"
+#include "DataFormats/ForwardDetId/interface/ForwardSubdetector.h"
+#include "DataFormats/ForwardDetId/interface/HGCalDetId.h"
+
+#include <vector>
+
+class Spot{
+    public:
+        Spot(DetId detid, double energy, const std::vector<double>& row, unsigned int layer, float fraction, double mip):
+            detId_(detid),energy_(energy),row_(row),layer_(layer),fraction_(fraction),
+            mip_(mip),multiplicity_(int(energy/mip)),subdet_(HGCalDetId(detid).subdetId()),isCore_(fraction>0.){;};
```

Comment:
is there really something non-trivial in ```HGCalDetId(detid).subdetId() ```?
It looks the same as ```detid.subdetId()``` 
IIRC, the latest proposals for HGC detid rework doesn't change this.

Reply:
@artlbv @beaudett is `HGCalDetId` construction necessary?